docs/AUTHORIZATION-ROADMAP.md

```diff
@@ -0,0 +1,20 @@
+# data.kitware.com Roadmap: Authorization, Permissions, User Groups
+
+Authorization settings, also called access control, for DKCN refers to both permissions and a public/private setting.
+
+DKCN will provide a permission system, and permissions will apply at a root folder level. DKCN will allow objects to have a public/private setting in addition to permissions. All folders and files descended from a root folder will have the same authorization settings as the root folder. If you want separate authorization settings, create a separate root folder.
+
+Authorizations are granted on an object to individual users or user groups, and the permission levels are: `read`, `write`, and `admin`. Read permission allows a user/group to read the object (including object properties, object metadata, and binary contents), write permission allows a user/group to mutate an object's properties except for permissions, admin permissions allow a user/group to see all the permissions on an object and to mutate the permissions on the object. Admin access implies write access, write access implies read access. If you do not have admin access to the object, you can see your own permissions and whether the object is public, but not the full permission set of the object.
+
+Users can have individual authorization to an object, or can have authorization to an object by virtue of being in a user group that has authorization to an object.
+
+Objects have a public/private setting. If an object is public, it acts as if it has read permission by the entire world, regardless of whether a user is even logged into the system. If an object is private, the object is not visible to a user unless the user has authorization to read the object via user or user group permissions.
+
+Currently the following design decisions exist:
+
+* Users are not access controlled.
+* User groups are not access controlled.
```

Comment:
These lines read like "users and groups are wide-open or publicly accessible". Since the actual access rules are described in the next few bullets, I'd recommend removing these two lines.